### ISSUE TITLE
Published Figures

### ISSUE THREAD

**Issue by tischi** [mobie/mobie-viewer-fiji#756]
Issue to keep track of the new "published figures" menu item.

Currently looks like this:

<img width="371" alt="image" src="https://user-images.githubusercontent.com/2157566/170109714-b2f18af6-b432-4376-b0c5-abb5021f9b49.png">

OK?

Or should we for now replace `Nature Methods` with `bioRxiv`?

**Comment by tischi** [comment]
- [ ] And we should probably also add the Cell Paper figures?!

**Comment by constantinpape** [comment]
Hmm, one thing: this can get very crowded very fast if we want to keep adding figures for multiple papers. So having a hierarchy that allows us to first select the paper and then the figure from the paper would make it simpler to navigate.

> Or should we for now replace `Nature Methods` with `bioRxiv`?

Yes, absolutely! We should not put Nature Methods on anything official yet.

> [ ] And we should probably also add the Cell Paper figures?!

Yea, good idea; but I wouldn't put it on highest priority.

**Comment by tischi** [comment]
> Hmm, one thing: this can get very crowded very fast if we want to keep adding figures for multiple papers.

I agree, but for now, maybe leave it as is. Nesting may be a bit complicated..

**Comment by tischi** [comment]
Updated MoBIE-beta to contain this feature.
Calling it a day now, will check again tomorrow afternoon for MoBIE stuff.

**Comment by constantinpape** [comment]
Typo in covid fixed here: https://github.com/mobie/mobie-viewer-fiji/pull/757
Besides that works for me :).

**Comment by constantinpape** [comment]
I think this functionality is working correctly. We can maybe add a few more figure panels, also from other papers, but as long as we don't have nesting I personally wouldn't do it (e.g. the platy paper would add 20 figures or so, which would make the menu very chaotic already.)

**Comment by tischi** [comment]
Yes, we have to think about how to implement the nesting...
@NicoKiaru do you know whether the SciJava commands support some TreeView?
Or some other way of having a nested drop-down?

**Comment by NicoKiaru** [comment]
Not directly I'm afraid. For the SourceAndConverter I've added some swing widgets:

https://github.com/bigdataviewer/bigdataviewer-playground/blob/master/src/main/java/sc/fiji/bdvpg/scijava/widget/SwingSourceAndConverterWidget.java

and 

https://github.com/bigdataviewer/bigdataviewer-playground/blob/master/src/main/java/sc/fiji/bdvpg/scijava/widget/SourceAndConverterWidget.java

I wrote this specific inputwidget because I was dealing with inputs of type `SourceAndConverter`. But what is your input type in this case ?

**Comment by tischi** [comment]
> But what is your input type in this case ?

Just a `Map< String, List<String> >`. Do you have something for this?

**Comment by NicoKiaru** [comment]
No, I do not have anything unfortunately. Maybe an idea to avoid having an input widget which takes all Map is to make a more specific class like:

```
class MobieTree<K,V> {
        Map<K,V> getMap();
}
```

and create an inputwidget which deals with MobieTree.

**Comment by tischi** [comment]
- [ ] Convert the internal data structure to a `Map< Publication==String, List< Figures==String >`
- [ ] Publications: `Platy`, `MoBIE`
